### DISCUSSION TITLE
Is it possible to merge OpenAPI document definitions?

### DISCUSSION THREAD

**Discussion by chanced** [juhaku/utoipa#608] [Q&A] [answered]
First off, thanks for the awesome work!

My API is spread out over numerous crates. Some of them have routers and components. The main application creates an axum router composed of nested routers.

I'd really like to keep the OpenAPI docs as close to their respective router as possible. Is it be possible to merge all of the definitions into a single unified OpenAPI document?

Thanks!

closing #607 as it should have been a discussion.

**Comment by juhaku** [comment, marked as answer]
Yes there is, you can create mulitple instances of ApiDoc types derived from `#[utoipa(OpenApi)]` to their respective crates and then merge all the them into one in main crate. There is a merge method in OpenApi for this purpose https://docs.rs/utoipa/latest/utoipa/openapi/struct.OpenApi.html#method.merge.

**Reply by chanced** [reply]
🤦‍♂️  sorry, i didn't see that.

Thank you!

**Reply by jayvdb** [reply]
There isnt many tests for `merge`, and IMO we should add an example that demonstrates this.  It is quite powerful when scaling up use of utoipa for multiple microservices.
